libpromises/evalfunction.c

```diff
@@ -585,41 +678,11 @@ static Rlist *GetHostsFromLastseenDB(Item *addresses, time_t horizon, bool retur
     }
 }
 
-/*********************************************************************/
-
-static FnCallResult FnCallAnd(EvalContext *ctx,
-                              ARG_UNUSED const Policy *policy,
-                              ARG_UNUSED const FnCall *fp,
-                              const Rlist *finalargs)
-{
-
-    for (const Rlist *arg = finalargs; arg; arg = arg->next)
-    {
-        SyntaxTypeMatch err = CheckConstraintTypeMatch(fp->name, arg->val, CF_DATA_TYPE_STRING, "", 1);
-        if (err != SYNTAX_TYPE_MATCH_OK && err != SYNTAX_TYPE_MATCH_ERROR_UNEXPANDED)
-        {
-            FatalError(ctx, "Function '%s', %s", fp->name, SyntaxTypeMatchToString(err));
-        }
-    }
-
-    for (const Rlist *arg = finalargs; arg; arg = arg->next)
-    {
-        if (!IsDefinedClass(ctx, RlistScalarValue(arg)))
-        {
-            return FnReturnContext(false);
-        }
-    }
-
-    return FnReturnContext(true);
-}
-
-/*******************************************************************/
-
```

Comment:
Please move FnCallAnd back up to where it was, to make a nicer diff. Then make a separate commit where you only move it.